### PR TITLE
add weather variable

### PR DESCRIPTION
**Pull request by dmtsolovey**

### CODE REVIEW

File: template.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+
+const apiUrl = "https://api.openweathermap.org/data/2.5/weather?";
+const units = data.units;
+const apiKey = data.apiKey;
+
+function getCity() {
+    const cityFromHeaders = getRequestHeader('X-Geo-City');
+    if(cityFromHeaders === 'ZZ' || !cityFromHeaders) {
+        return "We couldn't determine your city";
```
Comment (kHorozhanov) [inline template.js]: return null

File: template.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    } else {
+        return cityFromHeaders;
+    }
+}
+
+const city = getCity();
+
```
Comment (kHorozhanov) [inline template.js]: if !city return null

File: template.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+const url = apiUrl + "q=" + enc(city) + "&appid=" + enc(apiKey) + "&units=" + enc(units);
+return sendRequest(url);
+
+function sendRequest(url) {
+    return sendHttpRequest(url).then((successResult) => {
```
Comment (kHorozhanov) [inline template.js]: rename successResult to response

File: template.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+        if (successResult.statusCode === 301 || successResult.statusCode === 302) {
+            return sendRequest(successResult);
```
Comment (kHorozhanov) [inline template.js]: add logs like in Http varable

File: LICENSE [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+      To apply the Apache License to your work, attach the following
+      boilerplate notice, with the fields enclosed by brackets "[]"
+      replaced with your own identifying information. (Don't include
+      the brackets!)  The text should be enclosed in the appropriate
+      comment syntax for the file format. We also recommend that a
+      file or class name and description of purpose be included on the
+      same "printed page" as the copyright notice for easier
+      identification within third-party archives.
+
+   Copyright 2022 Stape
```
Comment (Bukashk0zzz) [inline LICENSE]: 2024

File: metadata.yaml [hunk 6]
```diff
@@ -0,0 +1,3 @@
+homepage: 'https://stape.io/'
```
Comment (Bukashk0zzz) [inline metadata.yaml]: do not commit this file before this PR will not merged.
remove this file from this PR

File: template.js [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+const getContainerVersion = require('getContainerVersion');
+
+
+const apiUrl = "https://api.openweathermap.org/data/2.5/weather?";
+const units = data.units;
+const apiKey = data.apiKey;
+const isLoggingEnabled = determinateIsLoggingEnabled();
+const traceId = isLoggingEnabled ? getRequestHeader('trace-id') : undefined;
+function getCity() {
+    const cityFromHeaders = getRequestHeader('X-Geo-City');
```
Comment (Bukashk0zzz) [inline template.js]: There need to be ability to choose in UI from with parameter get City and Country
By default get from Stape headers  but provide ability to override this.

File: template.js [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+    if(cityFromHeaders === 'ZZ' || !cityFromHeaders) {
+        return null;
+    } else {
+        return cityFromHeaders;
+    }
+}
+
+const city = getCity();
+if(!city) return null;
+const url = apiUrl + "q=" + enc(city) + "&appid=" + enc(apiKey) + "&units=" + enc(units);
```
Comment (Bukashk0zzz) [inline template.js]: look like it good Idea to provide country also as one city can be in multiple countries.

**Review by Bukashk0zzz — APPROVED** [review]

**Review by Bukashk0zzz — CHANGES_REQUESTED** [review]
Changes

File: metadata.yaml
Comment (Bukashk0zzz) [inline metadata.yaml:1]: Remove this file from PR

File: template.js [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+const getContainerVersion = require('getContainerVersion');
+
+
+const apiUrl = "https://api.openweathermap.org/data/2.5/weather?";
+const units = data.units;
+const apiKey = data.apiKey;
+const isLoggingEnabled = determinateIsLoggingEnabled();
+const traceId = isLoggingEnabled ? getRequestHeader('trace-id') : undefined;
+function getGeoInfo() {
+    const cityFromHeaders = getRequestHeader('X-Geo-City') || data.city;
```
Comment (Bukashk0zzz) [inline template.js]: data.city || getRequestHeader('X-Geo-City') change order.a

File: template.js [hunk 10]
```diff
@@ -0,0 +1,91 @@
...
+    const countryFromHeaders = getRequestHeader('X-Geo-Country') || data.countryCode || '';
```
Comment (Bukashk0zzz) [inline template.js]: data.countryCode || getRequestHeader('X-Geo-Country') change order

File: template.js [hunk 11]
```diff
@@ -0,0 +1,91 @@
...
+        return {
+            city: cityFromHeaders,
+            country: countryFromHeaders
+        };
+    }
+}
+
+const geo = getGeoInfo();
+if(!geo) return null;
+const geoCountry = geo.country ? ',' + geo.country : '';
```
Comment (Bukashk0zzz) [inline template.js:29]: You already checked at this moment if country exist

**Review by Bukashk0zzz — CHANGES_REQUESTED** [review]
@kHorozhanov review also.

**Pull request merged**